lesson4/scripts_calendar.js

```diff
@@ -0,0 +1,148 @@
+const selectors = {
+  prevMonth: "month_prev",
+  currMonth: "month_curr",
+  nextMonth: "month_next",
+};
+
+const months = [
+  "January",
+  "February",
+  "March",
+  "April",
+  "May",
+  "June",
+  "July",
+  "August",
+  "September",
+  "October",
+  "November",
+  "December",
+];
+
+class Calendar {
+  constructor() {
+    this.selectedMonth = new Date();
+  }
+
+  // RENDER
+
+  render() {
+    this.renderMonth();
+    this.addMonthEventListeners();
+    this.addDaysEventListener();
+  }
+
+  renderMonth() {
+    const prev = document.getElementById(selectors.prevMonth);
+    const curr = document.getElementById(selectors.currMonth);
+    const next = document.getElementById(selectors.nextMonth);
+
+    const date = new Date(this.selectedMonth);
+    const monthIndex = date.getUTCMonth();
+
+    curr.innerHTML = this.convertDateToMonthString(date);
+
+    date.setMonth(monthIndex - 1);
+    prev.innerHTML = this.convertDateToMonthString(date);
+
+    date.setMonth(monthIndex + 1);
+    next.innerHTML = this.convertDateToMonthString(date);
+  }
+
+  // ACTIONS
+
+  addMonthEventListeners() {
+    const prev = document.getElementById(selectors.prevMonth);
+    const next = document.getElementById(selectors.nextMonth);
+
+    prev.addEventListener("click", this.setSelectedMonth.bind(this, -1));
+    next.addEventListener("click", this.setSelectedMonth.bind(this, +1));
+  }
+
+  //CALENDAR DAYS
+
+  addDaysEventListener() {
+    const calendar = document.querySelector(".calendar_days");
+    let currentDate = new Date(this.selectedMonth);
+
+    function updateCalendar() {
+      calendar.innerHTML = "";
+
+      const firstDay = new Date(
+        currentDate.getFullYear(),
+        currentDate.getMonth(),
+        0
+      );
+      const lastDay = new Date(
+        currentDate.getFullYear(),
+        currentDate.getMonth() + 1,
+        0
+      ).getDate();
+      const startingDay = firstDay.getDay();
+
+      for (let i = 0; i < startingDay; i++) {
+        const emptyCell = document.createElement("div");
+        calendar.appendChild(emptyCell);
+      }
+
+      for (let day = 1; day <= lastDay; day++) {
+        const monthIndex = currentDate.getUTCMonth();
+        const year = currentDate.getFullYear();
+        const dayCell = document.createElement("div");
+
+        dayCell.classList.add("day-cell");
+        dayCell.textContent = day;
+        calendar.appendChild(dayCell);
+        dayCell.addEventListener("click", function () {
```

Comment:
Could u move out this event function. For example, u could create new method.
Also, try to use methods **call, apply, bind**